python/ray/_private/runtime_env/agent/runtime_env_agent.py

```diff
@@ -303,10 +303,18 @@ async def GetOrCreateRuntimeEnv(self, request):
             f"{request.serialized_runtime_env}."
         )
 
+        # Parsed runtime env config; cache to avoid re-computation for multiple times.
+        runtime_env_config: RuntimeEnvConfig = None
+
         async def _setup_runtime_env(
             runtime_env: RuntimeEnv,
         ):
-            runtime_env_config = RuntimeEnvConfig.from_proto(request.runtime_env_config)
+            global runtime_env_config
+            if not runtime_env_config:
+                runtime_env_config = RuntimeEnvConfig.from_proto(
```

Comment:
what if a next call has a different request, which contains a different runtime_env_config ?

Reply:
We are handling the same `request` at L299

Reply:
I see. instead of using global, can we just runtime_env_config in GetOrCreateRuntimeEnv, and pass it into _setup_runtime_env ?

Reply:
Unfortunately no, because on lucky path where runtime env is already cached, we don't need to parse `runtime_env_config`; we simply add reference count and return.

Reply:
OK I mis-understood, I will pass it as a function argument; please wait for me for a short while...

Reply:
Updated.